services/motion/server.go

```diff
@@ -61,6 +62,34 @@ func (server *serviceServer) MoveOnMap(ctx context.Context, req *pb.MoveOnMapReq
 	return &pb.MoveOnMapResponse{Success: success}, err
 }
 
+func (server *serviceServer) MoveOnGlobe(ctx context.Context, req *pb.MoveOnGlobeRequest) (*pb.MoveOnGlobeResponse, error) {
+	svc, err := server.coll.Resource(req.Name)
+	if err != nil {
+		return nil, err
+	}
+	obstaclesProto := req.GetObstacles()
+	obstacles := make([]*referenceframe.GeoObstacle, 0, len(obstaclesProto))
+	for _, eachProtoObst := range obstaclesProto {
+		convObst, err := referenceframe.GeoObstacleFromProtobuf(eachProtoObst)
+		if err != nil {
+			return nil, err
+		}
+		obstacles = append(obstacles, convObst)
+	}
+	success, err := svc.MoveOnGlobe(
+		ctx,
+		protoutils.ResourceNameFromProto(req.GetComponentName()),
+		geo.NewPoint(req.GetDestination().GetLatitude(), req.GetDestination().GetLongitude()),
+		req.GetHeading(),
```

Comment:
This will return `0` if not set on the request; we should manually do a `nil` check on the `Heading` field in the api struct, which will be of type `*float32`

Reply:
Yeah, I will add that now. I didn't want to spend too much time on the client/server code just yet, but since our only `optional` here is heading, no harm in just taking care of that.

Reply:
I think you should do this for the other optionals now too.